src/bin/purge_ttl.rs

```diff
@@ -132,21 +163,73 @@ fn main() -> Result<(), Box<dyn Error>> {
         {
             let _timer_batches = start_timer(&statsd, "purge_ttl.batches_duration");
             let mut req = prepare_request(&client, &session)?;
-            req.set_sql("DELETE FROM batches WHERE expiry < CURRENT_TIMESTAMP()".to_string());
-            let result = client.execute_sql(&req)?;
+            req.set_sql("SELECT (collection_id, fxa_kid, fxa_uid, batch_id) FROM batches WHERE expiry < CURRENT_TIMESTAMP()".to_string());
+            let result = SyncResultSet { result: client.execute_sql(&req)? };
+            let mut sql = "DELETE FROM batches WHERE (collection_id, fxa_kid, fxa_uid, batch_id) IN (".to_string();
+            let mut i = 0;
+            for row in result {
+                i = i + 1;
+                let collection_id = row[0].get_string_value().parse::<i64>().unwrap();
+                let fxa_kid = row[1].get_string_value().to_owned();
+                let fxa_uid = row[2].get_string_value().to_owned();
+                let batch_id = row[3].get_string_value().to_owned();
+                if i % BATCH_SIZE == 0 {
+                    sql = format!("{}({}, '{}', '{}', '{}'))", sql, collection_id, fxa_kid, fxa_uid, batch_id);
```

Comment:
use a different var name here? So you don't have to restore it later?

Reply:
This is outdated because clippy made me switch to enumerate and get rid of the relevant code.